x-pack/plugins/siem/public/store/inputs/epic.ts

```diff
@@ -43,22 +43,12 @@ export const createGlobalTimeEpic = <State>(): Epic<
       timer(0, policy.duration).pipe(
         map(() => {
           const option = get('option', timerange);
-          if (option === 'quick-select') {
-            const diff = timerange.to - timerange.from;
-            return setRelativeRangeDatePicker({
-              id: 'global',
-              option,
-              to: Date.now(),
-              from: moment()
-                .subtract(diff, 'ms')
-                .valueOf(),
-            });
-          }
+          const momentDate = dateMath.parse(option!);
```

Comment:
if the use of `!` to override type safety is protected by the `filter`, consider adding a comment like:

```
// the `filter` above on timerange.kind === 'relative' ensures `option` is never `undefined`
```

Reply:
and what about that `momentDate != null && momentDate.isValid() ? momentDate.valueOf() : 0`